test/http.test.lua

```diff
@@ -388,4 +413,81 @@ test:test("server requests", function(test)
+            return log
+        end
+    end
+end
+
+local function clear_log_queue()
+    log_queue = {}
+end
+
+test:test("Custom log level for route", function(test)
+    test:plan(3)
+    local httpd = http_server.new("127.0.0.1", 12345, { log_requests = true, log_errors = false })
+
+    test:test("Setting log_level for route", function(test)
+        test:plan(6)
+        local ok, err = pcall(httpd.route, httpd, { path = '/', log_level = 0 })
+
+        test:is(ok, false, "log_level should be in range [1;6]")
+        test:like(err, "'log_level' option should be a number in range from 1 to 6", "route() should return error message")
+
+        ok, err = pcall(httpd.route, httpd, { path = '/', log_level = 7 })
+        test:is(ok, false, "log_level should be a number in range [1;6]")
+        test:like(err, "'log_level' option should be a number in range from 1 to 6", "route() should return error message")
+
+        ok, err = pcall(httpd.route, httpd, { path = '/', log_level = 'info' })
+        test:is(ok, false, "log_level should be a number in range [1;6]")
+        test:like(err, "'log_level' option should be a number in range from 1 to 6", "route() should return error message")
+    end)
+
+    test:test("Log output with different log level on route", function(test)
+        test:plan(4)
+        httpd:start()
+        -- log level by default is 'INFO' (5)
+
+        httpd:route({ path = '/', log_level = 1 }, function(_) end)
+        http_client.get("127.0.0.1:12345")
+        test:isnil(find_msg_in_log_queue("GET /"), "Route should ignore logging requests on log_level = 1")
+        clear_log_queue()
+
+        httpd.routes = {}
+        httpd:route({ path = '/', log_level = 6 }, function(_) end)
+        http_client.get("127.0.0.1:12345")
+        test:isnil(find_msg_in_log_queue("GET /"), "Route should ignore logging requests if the root logger have less log_level")
+        clear_log_queue()
+
+        httpd.routes = {}
+        httpd:route({ path = '/', log_level = 4 }, function(_) end)
+        http_client.get("127.0.0.1:12345")
+        test:is_deeply(find_msg_in_log_queue("GET /"), { log_lvl = 'warn', msg = 'GET /' }, "Route should logging requests if the root logger have higher log_level")
+        clear_log_queue()
+
+        httpd.routes = {}
+        httpd:route({ path = '/', log_level = 3 }, function(_) end)
+        http_client.get("127.0.0.1:12345")
+        test:is_deeply(find_msg_in_log_queue("GET /"), { log_lvl = 'error', msg = 'GET /' }, "Route should logging requests if the root logger have higher log_level")
+        clear_log_queue()
+
+        httpd:stop()
+    end)
+
+    test:test("Log route requests with turned off 'log_requests' option", function(test)
+        test:plan(1)
+        local httpd_without_logs = http_server.new("127.0.0.1", 12345, { log_requests = false, log_errors = false })
+        httpd_without_logs:start()
+
+        httpd:route({ path = '/', log_level = 3 }, function(_) end)
```

Comment:
`httpd_without_logs:route` maybe?
After this fix the test should fail.

Reply:
yeah
you right. copy - paste is evil